pkg/crc/services/dns/dns_darwin.go

```diff
@@ -0,0 +1,50 @@
+package dns
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path/filepath"
+	"text/template"
+
+	"github.com/code-ready/crc/pkg/crc/services"
+)
+
+const (
+	resolverFileTemplate = `port {{ .Port }}
+	nameserver {{ .IP }}
+search_order {{ .SearchOrder }}`
+)
+
+type resolverFileValues struct {
+	Port        int
+	IP          string
+	SearchOrder int
+}
+
+func runPostStartForOS(serviceConfig services.ServicePostStartConfig, result *services.ServicePostStartResult) (services.ServicePostStartResult, error) {
+	// Write resolver file
+	createResolverFile(serviceConfig.HostIP, filepath.Join("/", "etc", "resolver", serviceConfig.BundleMetadata.ClusterInfo.BaseDomain))
+
+	result.Success = true
+	return *result, nil
+}
+
+func createResolverFile(hostIP string, path string) (bool, error) {
+	var resolverFile bytes.Buffer
+
+	values := resolverFileValues{
+		Port:        dnsServicePort,
+		IP:          hostIP,
+		SearchOrder: 1,
+	}
+
+	t, err := template.New("resolver").Parse(resolverFileTemplate)
+	if err != nil {
+		return false, err
+	}
+	t.Execute(&resolverFile, values)
+
+	err = ioutil.WriteFile(path, resolverFile.Bytes(), 0644)
```

Comment:
In mac this path required sudo permission to write in the file so we need to handle return error back.

```golang
+       if err := ioutil.WriteFile(path, resolverFile.Bytes(), 0644); err != nil {
+               return false, err
+       }
```

Reply:
that means this file needs to be created by the setup, and given permissions to the user to override or write to this folder.